include/ifc/abstract-sgraph.hxx

```diff
@@ -3544,7 +3546,20 @@ namespace ifc {
 
     // -- exception type in case of an invalid partition name
     struct InvalidPartitionName {
-        std::string_view name;
+        InvalidPartitionName(std::string_view partition_name)
+        {
+            partition_name = partition_name.substr(0, partition_name_buffer.size() - 1);
+            std::copy(partition_name.begin(), partition_name.end(), partition_name_buffer.begin());
+            partition_name_buffer[partition_name.length()] = 0;
```

Comment:
`partition_name` is already sized in such a way that the last byte in the buffer stays the way it was initiliazed, which is `0`. This line is unnecessary.